### PR TITLE
feat(compare-images): add spatial tolerance

### PR DESCRIPTION
**Pull request by jadh4v**
Add an option to pass spatial tolerance when comparing image direction matrix and origin for spatial overlap.

Also export `toScalarDouble` js function for external use.

### CODE REVIEW

File: packages/compare-images/typescript/src/index-node-only.ts [hunk 1]
```diff
@@ -18,3 +18,6 @@ export type { VectorMagnitudeNodeResult }
 
 import vectorMagnitudeNode from './vector-magnitude-node.js'
 export { vectorMagnitudeNode }
+
+import toScalarDouble from './to-scalar-double.js'
+export { toScalarDouble }
```
Comment (jadh4v) [inline packages/compare-images/typescript/src/index-node-only.ts:23]: Exporting `toScalarDouble` here for external use. This is useful to export test images as double and testing with native builds etc. May not be super crucial, but just for convenience.

File: packages/compare-images/typescript/src/compare-double-images.ts [hunk 2]
```diff
@@ -73,6 +73,10 @@ async function compareDoubleImages(
   if (typeof options.radiusTolerance !== "undefined") {
     args.push('--radius-tolerance', options.radiusTolerance.toString())
 
+  }
+  if (typeof options.spatialTolerance !== "undefined") {
+    args.push('--spatial-tolerance', options.spatialTolerance.toString())
+
```
Comment (jadh4v) [inline packages/compare-images/typescript/src/compare-double-images.ts:79]: manual update, file doesn't get auto-generated

File: packages/compare-images/typescript/src/compare-double-images-node.ts [hunk 3]
```diff
@@ -72,6 +72,10 @@ async function compareDoubleImagesNode(
   if (typeof options.radiusTolerance !== "undefined") {
     args.push('--radius-tolerance', options.radiusTolerance.toString())
 
+  }
+  if (typeof options.spatialTolerance !== "undefined") {
+    args.push('--spatial-tolerance', options.spatialTolerance.toString())
+
```
Comment (jadh4v) [inline packages/compare-images/typescript/src/compare-double-images-node.ts:78]: manual update, file doesn't get auto-generated

**Review by thewtex — CHANGES_REQUESTED** [review]
Thanks @jadh4v !

Minor request inline

File: packages/compare-images/python/itkwasm-compare-images/itkwasm_compare_images/compare_double_images.py [hunk 4]
```diff
@@ -12,6 +12,7 @@ def compare_double_images(
     test_image: Image,
     baseline_images: List[Image] = [],
     difference_threshold: float = 0,
+    spatial_tolerance: float = 1e-06,
```
Comment (thewtex) [inline packages/compare-images/python/itkwasm-compare-images/itkwasm_compare_images/compare_double_images.py:15]: again

File: packages/compare-images/python/itkwasm-compare-images/itkwasm_compare_images/compare_double_images_async.py [hunk 5]
```diff
@@ -12,6 +12,7 @@ async def compare_double_images_async(
     test_image: Image,
     baseline_images: List[Image] = [],
     difference_threshold: float = 0,
+    spatial_tolerance: float = 1e-06,
```
Comment (thewtex) [inline packages/compare-images/python/itkwasm-compare-images/itkwasm_compare_images/compare_double_images_async.py:15]: again

Comment (jadh4v) [reply packages/compare-images/python/itkwasm-compare-images/itkwasm_compare_images/compare_double_images_async.py]: also missed out on updating the wasm binary

**Review by thewtex — APPROVED** [review]
:100:

**Pull request merged**